tests/integration/tier0/check-user-response/test_user_response.py

```diff
@@ -0,0 +1,50 @@
+from envparse import env
+
+
+def test_check_user_response(convert2rhel):
+
+    # Run c2r registration with no username and password provided
+    # check for user prompt enforcing input, then continue with registration
+    with convert2rhel(
+        "-y --no-rpm-va --serverurl {}".format(
+            env.str("RHSM_SERVER_URL"),
+        )
+    ) as c2r:
+        c2r.expect_exact(" ... activation key not found, username and password required")
+        c2r.expect_exact("Username: ")
+        c2r.sendline()
+        if c2r.expect_exact(["Username", "Password"]) == 0:
+            # Provide username, expect password prompt
+            c2r.sendline(env.str("RHSM_USERNAME"))
+        else:
+            c2r.sendcontrol("d")
+        c2r.expect_exact("Password: ")
+        c2r.sendline()
+        if c2r.expect_exact(["Password", "Enter number of the chosen subscription: "]) == 0:
+            # Provide password, expect successful registration and subscription prompt
+            c2r.sendline(env.str("RHSM_PASSWORD"))
+        else:
+            c2r.sendcontrol("d")
+        c2r.expect_exact("Enter number of the chosen subscription: ")
+        c2r.sendcontrol("d")
+    assert c2r.exitstatus != 0
+
+    # Run c2r registration with activation key provided
+    # check for user prompt while organization left blank
+    with convert2rhel(
+        "-y --no-rpm-va --serverurl {} -k {}".format(
+            env.str("RHSM_SERVER_URL"),
+            env.str("RHSM_KEY"),
+        )
+    ) as c2r:
+        c2r.expect_exact(" ... activation key detected: ")
+        c2r.expect_exact("Organization: ")
+        c2r.sendline()
+        substitute_org = "foo"
+        if c2r.expect_exact("Organization: ") == 0:
+            c2r.sendline(substitute_org)
+        else:
+            c2r.sendcontrol("d")
+        c2r.expect_exact("Registering the system using subscription-manager ...")
+        c2r.sendcontrol("d")
```

Comment:
I would use `c` character to terminate the running process. Even though `ctrl d` may work I don't think it's safe to use it.

Reply:
I think in OL/CentOS 7 there is a bug where you have to hit Ctrl + C multiple times to exit the process... Don't know if it's a bug in how python handles it or it is a bug in our code, but I experienced it in here https://github.com/oamg/convert2rhel/pull/411

Reply:
![image](https://user-images.githubusercontent.com/2881357/153607636-0734235e-ab51-4f2e-ada9-2df16c514866.png)

The `^C` is for Ctrl + C and the `EOFError` at the end is for the actual Ctrl + D. 

(This is from CentOS 7)

Reply:
![image](https://user-images.githubusercontent.com/2881357/153608331-cd7afc5d-c713-4e37-84e9-13e41fb5de9e.png)

Works fine on CentOS 8 (Which is python3 by default)

Reply:
Oh I see now. If sending Ctrl + C doesn't work I would make a comment in code about it so it's clear why we use Ctrl + D instead.

Reply:
Totally agree! Can even link this thread to the comment in code

Reply:
Yeah, for some reason Ctrl + C doesn't work as consistently as Ctrl + D does.
Anyway I will add the comment mentioning this.